pkg/controller/proxyconfig/proxyconfig_controller.go

```diff
@@ -0,0 +1,206 @@
+	client client.Client
+	scheme *runtime.Scheme
+}
+
+func proxyNamespacedName() types.NamespacedName {
+	return types.NamespacedName{
+		Name: "cluster",
+	}
+}
+
+// Reconcile reads that state of the cluster for a cluslter-scoped
+// named "cluster" or a configmap object in namespace "openshift-config".
+func (r *ReconcileProxyConfig) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	if request.NamespacedName == proxyNamespacedName() {
+		proxyConfig := &configv1.Proxy{}
+		logrus.Debugf("Proxyconfig reconcile request.Name: '%s'", request.Name)
+		if err := r.client.Get(context.TODO(), request.NamespacedName, proxyConfig); err != nil {
+			if apierrors.IsNotFound(err) {
+				// Request object not found, could have been deleted after reconcile request.
+				// Return and don't requeue
+				logrus.Errorf("proxy not found; reconciliation will be skipped - request %v", request)
+				return reconcile.Result{}, nil
+			}
+			// Error reading the object - requeue the request.
+			return reconcile.Result{}, fmt.Errorf("failed to get proxy '%s': %v", request.Name, err)
+		}
+
+		if proxyConfig.Name != proxyName {
+			return reconcile.Result{}, nil
+		}
+	} else if request.Namespace == trustBundleConfigmapNS {
+		trustBundle := &corev1.ConfigMap{}
+		logrus.Debugf("Trust bundle configmap reconcile request.Namespace/request.Name: '%s/%s'", request.Namespace, request.Name)
+		if err := r.client.Get(context.TODO(), request.NamespacedName, trustBundle); err != nil {
+			if apierrors.IsNotFound(err) {
+				// Request object not found, could have been deleted after reconcile request.
+				// Return and don't requeue
+				logrus.Errorf("configmap not found; reconciliation will be skipped - request %v", request)
+				return reconcile.Result{}, nil
+			}
+			// Error reading the object - requeue the request.
+			return reconcile.Result{}, fmt.Errorf("failed to get configmap '%s': %v", request, err)
+		}
+
+		// Only proceed if request matches the configmap referenced by proxy trustedCA.
+		if err := r.configMapIsProxyTrustedCA(trustBundle.Name); err != nil {
+			logrus.Errorf("configmap '%s/%s' name differs from trustedCA of proxy '%s' or trustedCA not set; "+
+				"reconciliation will be skipped", trustBundle.Namespace, trustBundle.Name, proxyName)
+			return reconcile.Result{}, nil
+		}
+	} else {
+		return reconcile.Result{}, nil
+	}
+
+	// Fetch the ClusterLogging instance
+	instance := &loggingv1.ClusterLogging{}
+	if err := r.client.Get(context.TODO(), types.NamespacedName{Namespace: "openshift-logging", Name: singletonName}, instance); err != nil {
+		if apierrors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
```

Comment:
we don't want to requeue on an error, do we? if we do the following line doesn't requeue...